libs/common/src/components/osmosis-screen-container/index.tsx

```diff
@@ -9,22 +10,29 @@ import { OsmosisHeader, OsmosisHeaderProps } from "../osmosis-header";
 export interface OsmosisScreenContainerProps
   extends Omit<OsmosisHeaderProps, "onClose"> {
   children: ReactNode;
+  backgroundStyle?: BackgroundStyle;
 }
 
+const getBackGroundStyles = (style: BackgroundStyle) => {
+  return {
+    ...(style.color && { backgroundColor: style.color }),
+    ...(style.color && { backgroundSize: style.size }),
+    ...(style.color && { backgroundPosition: style.position }),
+    ...(style.color && { backgroundImage: `url(${style.image})` }),
+    ...(style.blendMode && { backgroundBlendMode: style.blendMode }),
+  };
+};
 export const OsmosisScreenContainer = observer<OsmosisScreenContainerProps>(
-  function OsmosisScreenContainer({ children, ...props }) {
+  function OsmosisScreenContainer({ children, backgroundStyle, ...props }) {
     const theme = useTheme();
     const onClose = useOnClose();
 
     return (
       <View
         style={{
           flex: 1,
-          // @ts-expect-error works in web
-          backgroundImage: theme.background.image
-            ? `url(${theme.background.image})`
-            : undefined,
-          backgroundColor: theme.background.color,
+          ...getBackGroundStyles(theme.background),
+          ...(backgroundStyle && getBackGroundStyles(backgroundStyle)),
```

Comment:
That's probably also not want you want. For example, this won't work well if `theme.background` contains an image, but `theme.welcome.background` doesn't. Instead: `getBackgroundStyle(backgroundStyle ?? theme.background)` doesn't leak css properties from unrelated theme properties.

Reply:
Also `getBackgroundStyle`, not `getBackGroundStyle`. Please be consistent in our naming.

Reply:
> Are you sure it won't work? I see osmosis working correctly

Because we don't have the edge case I mentioned. Doesn't change the fact that the code won't work correctly in these edge cases (which me might have in the future), and the code handling that edge case is shorter.

Reply:
let me see if I understand correctly, you mentioned a case where there is an image defined for the background but there is no image defined for `welcome.background`, that's the case for osmosis, and its working correctly

Reply:
<img width="1493" alt="Screenshot 2023-06-23 at 21 22 18" src="https://github.com/obi-wallet/obi-wallet-internal/assets/2956853/217baf10-457c-40ef-acdb-ab76bdeef663">
Look, osmosis background is in ` theme.background.image` and in this theme there is no `theme.welcome.background.image`, where in ztx it's the oposite.

Reply:
I mean, osmosis doesn't have `welcome.background`. With your current implementation, you couldn't design a theme where the welcome screen doesn't have a background image (but, for example, a different background color than other screens). Since the image property from `theme.background` would leak.